src/HTML5.php

```diff
@@ -234,6 +234,10 @@ public function saveHTML($dom, $options = array())
         $stream = fopen('php://temp', 'wb');
```

Comment:
`fopen` could return `false`

```php
        $stream = fopen('php://temp', 'wb');
        if ($stream === false) {
            // Acts accordingly I would throw an exception
            throw new \RuntimeException('Cannot create temporary stream');
        }

        try {
            $this->save($dom, $stream, array_merge($this->defaultOptions, $options));
            return stream_get_contents($stream, -1, 0);
        } finally {
          fclose($stream);
        }
```

Reply:
Very good catch. Would you mind to submit a pr ?
If you can't, no problem, I'll do it

Reply:
Sorry, I do not use this library and I am already occupied on other things. Feel free to submit a PR to fix it.

Btw, the same trick should be used in the above method where return from fopen is not checked for correctness.

I would split `save` method in with a private method `doSaveStream(...)` and `save` calling this directly whether `$file` is a resource or a filename.

Reply:
Arf, this lib should run on PHP 5.3 (yes :p )
I can not do that... I think we will keep this code like this

Reply:
```suggestion
        $stream = fopen('php://temp', 'wb');
        if ($stream === false) {
            // Acts accordingly I would throw an exception
            throw new \RuntimeException('Cannot create temporary stream');
        }

        try {
            $this->save($dom, $stream, array_merge($this->defaultOptions, $options));
            $result = stream_get_contents($stream, -1, 0);
        } catch (\Exception $exception) {
          fclose($stream);
          
          throw $exception;
        }
        
        fclose($stream);
        
        return $result;
```

Reply:
> Arf, this lib should run on PHP 5.3 (yes :p )
> I can not do that... I think we will keep this code like this

I think it would be preferable to update the library to require at least PHP 5.6.